code/datums/weather/weather.dm

```diff
@@ -192,11 +192,11 @@
 /datum/weather/proc/can_weather_act(mob/living/mob_to_check)
 	var/turf/mob_turf = get_turf(mob_to_check)
 
-	if(!(mob_turf.z in impacted_z_levels))
-		if(mob_turf)
-			return
+	if(mob_turf && !(mob_turf.z in impacted_z_levels))
+		return
 
-		if(!(mob_to_check.loc.z in impacted_z_levels)) // in this case, get_turf(mob_to_check) =/= mob_to_check.loc, so we have to proceed with checking the location's type
+	if(!(mob_turf) && mob_to_check.loc)
```

Comment:
you don't need the () around mob_turf here, since it's a normal check and not a list contents thing (We only use it for list contents because the "in" operator doesn't run first, so you end up checking if FALSE is in the list)